ocaml/xapi/xapi_pool_update.ml

```diff
@@ -110,35 +110,40 @@ let with_inc_refcount ~__context ~uuid ~vdi f =
         Hashtbl.replace updates_to_attach_count_tbl uuid (count + 1)
     )
 
+let get_locally_attached ~__context ~uuid ~vdi =
+    let mount_dir = String.concat "/" [!Xapi_globs.host_update_dir; uuid] in
+    let mount_dir_opt = if Sys.file_exists mount_dir then Some mount_dir else None in
```

Comment:
This could test for `Sys.is_directory` as well and avoid the test later.

Reply:
The test inside `detach_helper` was on the `vdi` sub-directory, this was on the parent directory. My previous patch did avoid re-testing the same directory.

But anyway, with the new patch, I've now left `detach_helper` completely untouched, as the refactoring seemed to have caused more confusion than benefits. Now, the semantic of `detach_helper` is just going ahead straightly and do the necessity same as before, leaving the judgement on any preconditions to its caller, which includes `detach`, `extract_update_info`, `detach_attached_updates` etc.